add radio-browser search integration

kgarner7 opened the pull request:
This is a draft integration of the [Radio Browser](https://de1.api.radio-browser.info/#Struct_station) API. 

It currently does:
- Periodic fetching of all station data from radio-browser (currently around 8.78 MB in DB)
- Allow searching through various filters
- Allow previewing a stream (playing it)
- Allow creating a new Internet Radio from the search. This also required additional fields in the `radio` table

Things missing:
- [ ] Translation. Most of the API is English, but it could be possible to translate the country code and language code?
- [x] [Click Metrics](https://de1.api.radio-browser.info/#Count_station_click). It is requested that clients using this API send a "Click" request when using a server. This is not yet implemented
-  ~~[ ] Proxy/store favicons? Some of them set third-party cookies, which isn't nice. Cons: extra load on Navidrome~~ (to be handled in another request)
- [ ] Better handling of non-functional streams. Because of CORS (as well as just being broken), some streams will just not play in the web client. I've added a warning, but it would be cool if there is some alternative

kgarner7 committed:
add radio-browser search integration

kgarner7 committed:
add cooldown for scan on start

kgarner7 committed:
make linter happy, click on button stop propagation

kgarner7 committed:
add error warning, better cleanup

deluan committed:
Simplify radio CRUD code

kgarner7 committed:
Merge branch 'master' into radio-search

kgarner7 committed:
merge simplification

kgarner7 committed:
forward clicks

kgarner7 committed:
merge master, refine appearance

deluan-tc commented on ui/src/App.js:
```diff
@@ -110,6 +111,10 @@ const Admin = (props) => {
+        permissions === 'admin' ? (
+          <Resource name="radioInfo" {...radioInfo} />
+        ) : undefined,
+        config.devEnableShare && <Resource name="share" {...share} />,
```
Merge error?

deluan-tc reviewed (changes requested):
Hey @kgarner7 why this is still a draft? Anyway, check my comments. Thanks! 

Re: "Translation. Most of the API is English, but it could be possible to translate the country code and language code?", I think it makes sense. We could maybe add the country code in the translation files, and get the translations from there.

kgarner7 committed:
Address PR

- Merge in master
- `/radio` => `/radios`. For some reason `/radio/${id}` just gave me a status=0? If the original one works that'd be great
- Move radio search to empty page/radio list
- Edit button for existing entries instead

kgarner7 committed:
Merge branch 'master' into radio-search

kgarner7 commented:
This was still in draft mainly for the translations (and in the hope that proxying radios would be looked at first). One weird thing I noticed when testing, for some reason `/radio/${id}/` would return with status 0, which is why I changed to `/radios`. If you can fix that it'd be great. I have no idea why the rest api was not called .-.

deluan commented:
> and in the hope that proxying radios would be looked at first

Do you think the radio proxy PR should go first?
